RFS/src/main/java/com/rfs/worker/DocumentsRunner.java

```diff
@@ -1,74 +1,81 @@
+                    @Override
+                    public Void onAlreadyCompleted() throws IOException {
+                        return null;
+                    }
 
-    @Override
-    public void runInternal() {
-        WorkerStep nextStep = null;
-        try {
-            nextStep = new DocumentsStep.EnterPhase(members);
+                    @Override
+                    public Void onAcquiredWork(IWorkCoordinator.WorkItemAndDuration workItem) throws IOException {
+                        doDocumentsMigration(IndexAndShard.valueFromWorkItemString(workItem.getWorkItemId()));
+                        return null;
+                    }
+                });
+    }
 
-            while (nextStep != null) {
-                nextStep.run();
-                nextStep = nextStep.nextStep();
-            }
-        } catch (Exception e) {
-            throw new DocumentsMigrationPhaseFailed(
-                members.globalState.getPhase(), 
-                nextStep, 
-                members.cmsEntry.map(bar -> (CmsEntry.Base) bar), 
-                e
-            );
+    public static class ShardTooLargeException extends RfsException {
+        public ShardTooLargeException(long shardSizeBytes, long maxShardSize) {
+            super("The shard size of " + shardSizeBytes + " bytes exceeds the maximum shard size of " + maxShardSize + " bytes");
         }
     }
 
-    @Override
-    public String getPhaseName() {
-        return "Documents Migration";
-    }
+    private void doDocumentsMigration(IndexAndShard indexAndShard) {
+        log.info("Migrating docs for " + indexAndShard);
+        ShardMetadata.Data shardMetadata = shardMetadataFactory.apply(indexAndShard.indexName, indexAndShard.shard);
 
-    @Override
-    public Logger getLogger() {
-        return logger;
-    }
+        try (var unpacker = unpackerFactory.create(shardMetadata)) {
+            var reader = readerFactory.apply(unpacker.unpack());
+            Flux<Document> documents = reader.readDocuments();
 
-    public static class DocumentsMigrationPhaseFailed extends Runner.PhaseFailed {
-        public DocumentsMigrationPhaseFailed(GlobalState.Phase phase, WorkerStep nextStep, Optional<CmsEntry.Base> cmsEntry, Exception e) {
-            super("Documents Migration Phase failed", phase, nextStep, cmsEntry, e);
+            final int finalShardId = shardMetadata.getShardId(); // Define in local context for the lambda
+            reindexer.reindex(shardMetadata.getIndexName(), documents)
+                    .doOnError(error -> log.error("Error during reindexing: " + error))
+                    .doOnSuccess(done -> log.info("Reindexing completed for Index " + shardMetadata.getIndexName() +
+                                    ", Shard " + finalShardId))
+                    // Wait for the reindexing to complete before proceeding
+                    .block();
+            log.info("Docs migrated");
```

Comment:
Is this needed with the doOnSuccess on 74? If yes, it feels like it should include some additional context in the log output.

Reply:
Probably not, but that's what was in the previous incarnations of the code, so I'll leave it for now to make sure that nobody else is expecting that line to be present.